znh5md/serialization/base.py

```diff
@@ -233,10 +238,16 @@ def __getitem__(self, idx: int) -> ase.Atoms:
         # this raises the IndexError to determine the length of the Frames object
         atoms = ase.Atoms(
             numbers=self.numbers[idx],
-            positions=self.positions[idx],
-            cell=self.cell[idx],
-            pbc=self.pbc[idx],
+            # positions=self.positions[idx],
+            # cell=self.cell[idx],
+            # pbc=self.pbc[idx],
         )
+        if len(self.positions) > 0:
+            atoms.set_positions(self.positions[idx])
+        if len(self.cell) > 0:
+            atoms.set_cell(self.cell[idx])
+        if len(self.pbc) > 0:
+            atoms.set_pbc(self.pbc[idx])
```

Comment:
might want to check the current frame. Also check if it would be faster to collect them in one dict and pass them to the ase constructor for better performance